app/crds.py

```diff
@@ -123,7 +123,11 @@ class ResourceSpec(BaseModel):
     security_policy_id: str | None = None
     is_visible: bool = True
     is_browser_shortcut_enabled: bool = True
-    protocols: ResourceProtocols | None = None
+    protocols: ResourceProtocols = ResourceProtocols(
```

Comment:
Added default to replicate API behavior - when no protocols limitation is set, reading resource from the API will return a protocols object like this (and not `null`)